subprojects/testing-native/src/main/java/org/gradle/nativeplatform/test/xctest/plugins/XCTestConventionPlugin.java

```diff
@@ -94,14 +95,20 @@ public void apply(ProjectInternal project) {
             @Override
             public void execute(Project project) {
                 ToolChainSelector.Result<SwiftPlatform> result = toolChainSelector.select(SwiftPlatform.class);
+                SwiftLanguageVersion swiftLanguageVersion = testSuite.getSwiftLanguageVersionSupport().getOrNull();
+                if (swiftLanguageVersion == null) {
+                    swiftLanguageVersion = SwiftLanguageVersion.of(result.getPlatformToolProvider().getCompilerMetadata().getVersion());
```

Comment:
I think we should default the swift version of the test suite to the version of the component under test, as a convention.

Also: more duplication.

Reply:
Good idea, let me see what I can do about this.